### PR TITLE
LPS-193771 - Create Deletion Not Allowed modal

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://liferay.atlassian.net/browse/LPS-193771

### CODE REVIEW

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -20381,6 +20381,7 @@ x-is-a-reserved-word={0} is a reserved word and cannot be used.
 x-is-allowed-to-do-action-x-in-all-items-of-type-x-in-this-portal-instance={0} is allowed to do action "{1}" in all items of type {2} in this portal instance. Use the roles administration tool if you want to change this.
 x-is-allowed-to-do-action-x-in-all-items-of-type-x-in-x={0} is allowed to do action <em>{1}</em> in all items of type {2} in {3}. Use the roles administration tool if you want to change this.
 x-is-already-the-name-of-an-icon-pack="{0}" is already the name of an icon pack.
+x-is-being-used-by-a-root-object-and-cannot-be-deleted-to-delete-x-you-must-first-unbind-it={0} is being used by a Root Object and cannot be deleted. To delete {0}, you must first unbind it from the Root Object that is using it.
```
Comment (naraandrade) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:20384]: @danvb112 To follow our current pattern, the language key should be written considering just the first sentence of its value.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalDeletionNotAllowed.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayModal, {useModal} from '@clayui/modal';
+import {sub} from 'frontend-js-web';
+import React from 'react';
+
+interface ModalDeletionNotAllowedProps {
+	onvisibilityChange: () => void;
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalDeletionNotAllowed.tsx:12]: @danvb112 Same comment as in the others PRs.

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Review by naraandrade — APPROVED** [review]

File: modules/apps/object/object-js-components-web/types/src/main/resources/META-INF/resources/utils/api.d.ts [hunk 3]
```diff
@@ -62,6 +62,7 @@ export interface NotificationTemplate {
 declare type ObjectRelationshipType = 'manyToMany' | 'oneToMany' | 'oneToOne';
 interface ObjectRelationship {
 	deletionType: string;
+	treeEdge: boolean;
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-js-components-web/types/src/main/resources/META-INF/resources/utils/api.d.ts:65]: hey @danvb112 could you please sort this parameter?

Thanks!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx [hunk 4]
```diff
@@ -32,6 +32,7 @@ import {ModalEditFolder} from './ModalEditFolder';
 import {deleteObjectDefinition, getFolderActions} from './objectDefinitionUtil';
 
 import './ViewObjectDefinitions.scss';
+import {ModalDeletionNotAllowed} from '../ModalDeletionNotAllowed';
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx:35]: @danvb112 I'm not sure, but I believe we should sort this import as well, what do you think?

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by danvb112** [comment]
ci:test:sf

File: modules/apps/object/object-js-components-web/types/src/main/resources/META-INF/resources/utils/api.d.ts (hunk 3 above)
Comment (danvb112) [reply modules/apps/object/object-js-components-web/types/src/main/resources/META-INF/resources/utils/api.d.ts:65]: Thanks for having notice that Feliphe! Done!

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/liferay-objects/liferay-portal/pull/2508

**Pull request closed without merging**